src/main/java/swinglib/JLabelBuilder.java

```diff
@@ -35,7 +37,10 @@ public JLabel build() {
     Preconditions.checkNotNull(text);
     Preconditions.checkState(!text.trim().isEmpty());
 
-    final JLabel label = new JLabel(text);
+    final String truncated = maxTextLength  > 0 && text.length() > maxTextLength ?
+        text.substring(0, maxTextLength) + "..." : text;
```

Comment:
Just noting that Apache Commons Lang has several `StringUtil#abbreviate()` methods that may be useful here.